### PR TITLE
RED-477 support bonfida sol

### PR DESCRIPTION
**Pull request by au-e2xlabs**

### CODE REVIEW

File: src/redefined.resolver.ts [hunk 1]
```diff
@@ -112,4 +114,8 @@ export class RedefinedResolver {
     static createSidArbNovaResolver(node?: string) {
         return new SidResolverService(node || config.SID_ARB_ONE_NODE, SidChainId.ARB, "arbitrum-nova", SidResolverData.ARB_NOVA);
     }
+
+    static createBonfidaResolver() {
```
Comment (Ivan-dev-RED) [inline src/redefined.resolver.ts:118]: @Rednoll 
Perhaps we need to allow the user to specify a cluster

File: src/services/resolvers/bonfida-resolver.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    constructor(
+        public cluster: Cluster = "mainnet-beta",
+    ) {
+        super();
+    }
+
+    async resolve(domain: string): Promise<Account[]> {
+        try {
+            const {pubkey} = getDomainKeySync(domain);
```
Comment (Ivan-dev-RED) [inline src/services/resolvers/bonfida-resolver.service.ts:22]: Minor:
add spaces `{ pubkey }`

File: src/services/resolvers/bonfida-resolver.service.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            const connection = SolWeb3Service.getWeb3(this.cluster);
```
Comment (Ivan-dev-RED) [inline src/services/resolvers/bonfida-resolver.service.ts:23]: I think we can initialize it in constructor

File: src/services/resolvers/bonfida-resolver.service.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+            const {registry} = await NameRegistryState.retrieve(connection, pubkey);
```
Comment (Ivan-dev-RED) [inline src/services/resolvers/bonfida-resolver.service.ts:25]: What if the domain is:
asd@asd
asd=-_!

What happens if the domain is not registered?

File: src/services/resolvers/bonfida-resolver.service.ts (hunk 3 above)
Comment (au-e2xlabs) [reply src/services/resolvers/bonfida-resolver.service.ts:23]: done

File: src/services/resolvers/bonfida-resolver.service.ts (hunk 2 above)
Comment (au-e2xlabs) [reply src/services/resolvers/bonfida-resolver.service.ts:22]: done

File: src/redefined.resolver.ts (hunk 1 above)
Comment (au-e2xlabs) [reply src/redefined.resolver.ts]: done

**Review by Ivan-dev-RED — APPROVED** [review]

File: tests/unit/services/resolvers/bonfida-resolver.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('bonfida-resolver.service', () => {
+    const bonfidaResolverService = new BonfidaResolverService();
+
+    const spyRetrieve = jest.spyOn(NameRegistryState, 'retrieve');
+
+    beforeEach(() => {
+        spyRetrieve.mockReset();
+    })
+
```
Comment (Rednoll) [inline tests/unit/services/resolvers/bonfida-resolver.service.spec.ts:27]: Add following tests:
- SHOULD does not get addresses for domain with network IF is invalid
- SHOULD correct handle errors

**Review by Rednoll — APPROVED** [review]

**Pull request merged**